app/models/course_user.rb

```diff
@@ -36,7 +36,12 @@ class CourseUser < ActiveRecord::Base
                                        inverse_of: :course_user, dependent: :destroy
   has_many :course_user_achievements, class_name: Course::UserAchievement.name,
                                       inverse_of: :course_user, dependent: :destroy
-  has_many :achievements, through: :course_user_achievements, class_name: Course::Achievement.name
+  has_many :achievements, through: :course_user_achievements,
+                          class_name: Course::Achievement.name do
+    def ordered_by_date_obtained
+      order('course_user_achievements.obtained_at desc')
```

Comment:
Because this is a calculated attribute, AR doesn't allow me to do it in this format, and I have to manually specify it. Will this be okay, or? 

This is the same problem as the scope `ordered_by_experience_points`.